front/src/applications/operationalStudies/hooks/usePathProjection.ts

```diff
@@ -78,17 +89,70 @@ const usePathProjection = (
 
   const projectingOnSimulatedPathException = pathfindingResultsDiffer(basePacedPath, pacedPath);
 
+  const pathUsedForProjection = useMemo(() => {
+    if (!trainIdUsedForProjection) return undefined;
+    if (!isOccurrenceId(trainIdUsedForProjection)) {
+      return timetableItemsById.get(trainIdUsedForProjection)?.path;
+    }
+    const pacedTrain = timetableItemsById.get(
+      extractPacedTrainIdFromOccurrenceId(trainIdUsedForProjection)
+    );
+    const exception = getExceptionFromOccurrenceId(timetableItemsById, trainIdUsedForProjection);
+    return exception?.path_and_schedule?.path ?? pacedTrain!.path;
+  }, [trainIdUsedForProjection, timetableItemsById]);
+
+  const operationalPointReferences: OperationalPointReference[] = [];
+  pathUsedForProjection?.forEach((step) => {
+    if (isOperationalPointReference(step)) {
+      const { id: _id, deleted: _deleted, ...cleanOperationalPointReference } = step;
+      operationalPointReferences.push(cleanOperationalPointReference);
+    }
+  });
+
+  const { data: operationalPoints } =
+    osrdEditoastApi.endpoints.postInfraByInfraIdMatchOperationalPoints.useQuery(
+      operationalPointReferences.length > 0
+        ? {
+            infraId,
+            body: {
+              operational_point_references: operationalPointReferences,
+            },
+          }
+        : skipToken
+    );
+
   return useMemo(() => {
     if (pathfinding?.status !== 'success' || !pathProperties) {
-      return undefined;
+      // Pathfinding failed: normalize operational points from frontend matching
+      const rawOps =
+        operationalPoints?.related_operational_points
+          .filter((ops) => ops.length !== 0) // To remove empty arrays related to invalid step
+          .map((matchingOps) => getStationFromOps(matchingOps)!) ?? [];
+
+      const normalizedOps: ProjectionOperationalPoint[] = rawOps.map((op, index) => ({
+        id: op.id,
+        extensions: op.extensions,
+        part: op.parts[0] || { track: '', position: 0, extensions: undefined },
```

Comment:
nit

```suggestion
        part: op.parts.at(0) || { track: '', position: 0, extensions: undefined },
```
But is it even possible to have an empty array ? Wouldn't this field have been optional if it was the case ? Or did you encounter that issue ?